static/js/markdown_config.js

```diff
@@ -0,0 +1,54 @@
+/*
+    This config is in a separate file for partly
+    tactical reasons.  We want the webapp to
+    configure this one way, but we don't want to
+    share this code with mobile.
+
+    I also wanted to make some diffs clear before
+    doing any major file moves.
+
+    Also, I want the unit tests for markdown to
+    be able to reuse this code easily (and therefore
+    didn't just put this in ui_init.js).
+
+    Once the first steps of making markdown be a
+    shared library are complete, we may tweak
+    the file organization a bit.
+*/
+
+exports.get_helpers = () => {
+    return {
+        // HTML escaping
+        escape: _.escape,
+
+        // user stuff
+        get_user_by_name: people.get_by_name,
+        get_user_by_id: people.get_by_user_id,
+        is_known_user_id: people.is_known_user_id,
```

Comment:
This can be replaced with `get_by_user_id(…) !== undefined`.

Reply:
The original motivation for `is_known_user_id` here is that we could avoid spammy blueslip errors from the subsequent call to `get_by_user_id`.  Unlike most places in the code, trying to get a person based on a user_id from our duplicate-names-are-possible-so-put-ids-in-the-mention syntax and then failing to get a user doesn't mean that we've broken some invariant in the code; i.e. it's just bad data from users (who may have editing the mention after autocomplete).

We now have an `ignore_missing` parameter in `get_by_user_id`, though, so there's really no need for `known_user_id`.

Here's the relevant code for context:

~~~ js
103         userMentionHandler: function (name, silently) {
104             let person = helpers.get_user_by_name(name);
105
106             const id_regex = /(.+)\|(\d+)$/g; // For @**user|id** syntax
107             const match = id_regex.exec(name);
108             if (match) {
109                 const user_id = parseInt(match[2], 10);
110                 if (helpers.is_known_user_id(user_id)) {
111                     person = helpers.get_user_by_id(user_id);
112                     if (person.full_name !== match[1]) { // Invalid Syntax
113                         return;
114                     }
115                 }
116             }
~~~

It turns out this code also has issues that @gnprice brought up in chat, though, which is that we shouldn't be asking mobile for "person" objects.  Instead, we should just have something `maybe_get_full_name_for_user_id`.

Anyway, I will address this.

Reply:
Also, that entire function is a mess. Ugh.